### PR TITLE
Add MiRFM fleet adapter package

### PR DESCRIPTION
**Pull request by xiyuoh**
This PR provides an additional fleet adapter package containing scripts and APIs that help to bridge RMF with the MiRFM instead of individual MiR robots. 
- Moves existing `fleet_adapter_mir` items into its own package, and an additional `fleet_adapter_mirfm` as a separate package
- `mir_fleet_client` package containing useful API for the MiRFleet
- Sample `mirfm_config.yaml` provided in the configs folder, but as usual it will have to be updated with the MiRFleet's base URL, user/password, as well as coordinate transformations under the `reference_coordinates` section.

### CODE REVIEW

File: configs/mirfm_config.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# ROBOT CONFIG =================================================================
+# To init MiR robot REST APIs and RMF states
+
+mirfm:
+  base_url: 'http://some.ip.address/api/v2.0.0'
```
Comment (koonpeng) [inline configs/mirfm_config.yaml:5]: Nit: Technically it supports url as well.

File: fleet_adapter_mirfm/test/test_flake8.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ament_flake8.main import main_with_errors
+import pytest
+
+
+@pytest.mark.flake8
+@pytest.mark.linter
+def test_flake8():
```
Comment (koonpeng) [inline fleet_adapter_mirfm/test/test_flake8.py:21]: Are these tests actually being used? If so, github actions should be added to run them, else they should be removed.

**Review by koonpeng — CHANGES_REQUESTED** [review]

File: fleet_adapter_mirfm/test/test_flake8.py (hunk 2 above)
Comment (xiyuoh) [reply fleet_adapter_mirfm/test/test_flake8.py]: You're right they were automatically generated, removed in [520d2ca](https://github.com/osrf/fleet_adapter_mir/pull/27/commits/520d2cab116e91ee03871b0263157cc94e0e5b33)

File: configs/mirfm_config.yaml (hunk 1 above)
Comment (xiyuoh) [reply configs/mirfm_config.yaml]: [520d2ca](https://github.com/osrf/fleet_adapter_mir/pull/27/commits/520d2cab116e91ee03871b0263157cc94e0e5b33)

File: configs/mirfm_config.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # To init MiR robot REST APIs and RMF states
 
 mirfm:
-  base_url: 'http://some.ip.address/api/v2.0.0'
```
Comment (koonpeng) [inline configs/mirfm_config.yaml]:
```suggestion
  base_url: 'http://mirfm_ip_or_url/api/v2.0.0'
```

**Review by koonpeng — CHANGES_REQUESTED** [review]

File: configs/mirfm_config.yaml (hunk 3 above)
Comment (xiyuoh) [reply configs/mirfm_config.yaml]: [8b96324](https://github.com/osrf/fleet_adapter_mir/pull/27/commits/8b96324eda9df882bae96d83e9874aa1683290aa)

**Review by koonpeng — APPROVED** [review]

**Pull request merged**